include/swift/AST/Types.h

```diff
@@ -577,6 +577,10 @@ class alignas(1 << TypeAlignInBits) TypeBase
 
   bool isPlaceholder();
 
+  /// Returns true if this is a move only type. Returns false if this is a
+  /// non-move only type or a move only wrapped type.
+  bool isPureMoveOnly() const;
```

Comment:
What does "pure" mean here? Why not just "isMoveOnly"? Why treat MoveOnlyWrapped types specially?